RTNeural/activation/activation_eigen.h

```diff
@@ -464,6 +464,418 @@ class PReLUActivationT
+        : Activation<T>(size, {}, "softplus")
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    SoftplusActivation(std::initializer_list<int> sizes)
+        : SoftplusActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outVec = (1 + MathsProvider::exp(inVec.array())).log(); //log() awaiting mathsprovider
+
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+    Eigen::Matrix<T, Eigen::Dynamic, 1> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, 1> outVec;
+};
+
+/** Static implementation of a Softplus activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class SoftplusActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>;
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    SoftplusActivationT()
+        : outs(outs_internal)
+    {
+        outs = v_type::Zero();
+    }
+
+    std::string getName() const noexcept { return "softplus"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    /** Forward propagation for Softplus activation. */
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outs = (1 + MathsProvider::exp(ins.array()).log(); //log() waiting for mathsprovider
+    }
+
+    Eigen::Map<v_type, RTNeuralEigenAlignment> outs;
+
+private:
+    T outs_internal alignas(RTNEURAL_DEFAULT_ALIGNMENT)[out_size];
+};
+
+/** Dynamic implementation of a Mish activation layer. */
+template <typename T, typename MathsProvider = DefaultMathsProvider>
+class MishActivation : public Activation<T>
+{
+public:
+    /** Constructor for Mish activation layer depending on the softplus activation internally */
+    explicit MishActivation(int size)
+        : Activation<T>(size, {}, "mish"), softplus(size)
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    MishActivation(std::initializer_list<int> sizes)
+        : MishActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+        /** mish = tanh(softplus(x)) */
+        Eigen::Matrix<T, Eigen::Dynamic, 1> softplusOut(Layer<T>::in_size);
```

Comment:
This is not real-time safe (I think the real-time sanitizer will catch it in CI, we'll see shortly). We can make this a member variable, so that the memory is allocated at construction-time.